src/ai/backend/manager/types.py

```diff
@@ -56,3 +70,134 @@ class MountOptionModel(BaseModel):
+
+    @classmethod
+    def set(cls, attr_name: str, value: TVal) -> Self:
+        return cls(attr_name, state=UpdateState.UPDATE, value=value)
+
+    @classmethod
+    def unset(cls, attr_name: str) -> Self:
+        return cls(attr_name, state=UpdateState.NULLIFY, value=None)
+
+    @classmethod
+    def nop(cls, attr_name: str) -> Self:
+        return cls(attr_name, state=UpdateState.NOP, value=None)
+
+    @property
+    def state(self) -> UpdateState:
+        return self._state
+
+    def value(self) -> TVal | None:
+        if self._state == UpdateState.UPDATE:
+            return self._value
+        raise ValueError(f"Value is not set for {self._attr_name}")
+
+    def set_attr(self, obj: Any) -> None:
+        match self._state:
+            case UpdateState.UPDATE:
+                setattr(obj, self._attr_name, self._value)
+            case UpdateState.NULLIFY:
+                setattr(obj, self._attr_name, None)
+            case UpdateState.NOP:
+                pass
+
+
+class NonNullState(TriState[TVal]):
+    _attr_name: str
+    _state: UpdateState
+    _value: TVal | None
+
+    def __init__(
+        self,
+        attr_name: str,
+        state: UpdateState,
+        value: TVal | None,
+    ) -> None:
+        self._attr_name = attr_name
+        self._state = state
+        self._value = value
+
+    @classmethod
+    def set(cls, attr_name: str, value: TVal) -> Self:
+        return cls(attr_name, state=UpdateState.UPDATE, value=value)
+
+    @classmethod
+    def none(cls, attr_name: str) -> Self:
+        return cls(attr_name, state=UpdateState.NOP, value=None)
+
+
+class TriStateField(Generic[TVal]):
+    _value: Optional[TVal] | Sentinel
+
+    def __init__(self, value: Optional[TVal] | Sentinel = SENTINEL) -> None:
+        self._value = value
+
+    def value(self) -> Optional[TVal]:
+        if self._value is not SENTINEL:
+            return self._value
+        raise ValueError(f"Value is not set for {self.__class__.__name__}")
+
+    @classmethod
+    def set(cls, value: TVal) -> Self:
+        return cls(value)
+
+    @classmethod
+    def unset(cls) -> Self:
+        return cls(None)
+
+    @classmethod
+    def nop(cls) -> Self:
+        return cls(SENTINEL)
+
+
+NonNullValue: TypeAlias = TVal | Sentinel
+
+
+class NonNullStateField(TriStateField[TVal]):
+    _value: TVal | Sentinel
+
+    def __init__(self, value: TVal | Sentinel = SENTINEL) -> None:
+        self._value = value
+
+    def value(self) -> TVal:
+        if self._value is not SENTINEL and self._value is not None:
+            return self._value
+        raise ValueError(f"Value is not set for {self.__class__.__name__}")
+
+    @classmethod
+    def set(cls, value: TVal) -> Self:
+        return cls(value)
+
+
+class TriStateData(UserDict):
+    def set_attr(self, obj: Any) -> None:
+        for field_name, value in self.data.items():
+            if value is not SENTINEL:
+                setattr(obj, field_name, value)
```

Comment:
I didn't get how to use `TriState` and `NonNullState` (also known as `UpdateOnlyState`) so I added this types.
[Here is how they are used](https://github.com/lablup/backend.ai/blob/406183543b665ff0bc2c639282dd8c0ea1017f00/src/ai/backend/manager/api/vfolder.py#L999) and please leave a review.